### ISSUE TITLE
TestTorch.test_doc should be in TestDocCoverage

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#28033]
Afaict TestTorch.test_doc tests that certain functions on torch are documented. This is similar to what TestDocCoverage does and it would be great to have all of the documentation tests in one single location.

https://github.com/pytorch/pytorch/blob/fd3d6587e60de816106d6b8dc0f586f9fa52a7ac/test/test_torch.py#L183

**Comment by ayush-1506** [comment]
@zou3519 I'd like to work on this issue, if that's not a problem? Please let me know if I should go ahead.

**Comment by ezyang** [comment]
Sure! However, I am not sure we are quite clear what exactly should be done here (unless @zou3519 you have a better idea than me). I suggest reading the code and then writing up a short proposal here that we can check over before you start coding.

**Comment by ayush-1506** [comment]
From what I could understand, the overall objective is to move the [test_doc](https://github.com/pytorch/pytorch/blob/fd3d6587e60de816106d6b8dc0f586f9fa52a7ac/test/test_torch.py#L183) to a member function inside TestDocCoverage ([here](https://github.com/pytorch/pytorch/blob/master/test/test_docs_coverage.py#L15)), since testing for documented functions is the purpose of both these.

At the moment I don't see any caveats, but given that I've merely skimmed over the code, I'd be grateful if someone could verify my understanding and let me know if I've missed something.

cc @ezyang @zou3519

**Comment by ayush-1506** [comment]
Ping @ezyang @zou3519

**Comment by zou3519** [comment]
When I opened this issue, I was thinking that we should consolidate all of our docs tests into one test file. i.e., `TestTorch.test_doc` and `TestTorch.test_doc_template` should be moved into the same test file as TestDocsCoverage.

The reason for this is to consolidate our docs tests and to perhaps share a blacklist/whitelist between them. Perhaps we should move `TestTorch.test_doc` and `TestTorch.test_doc_template` into a new TestCase that is named something like "TestDocstrings" and toss it into test_docs_coverage.py?

**Comment by ayush-1506** [comment]
Right, that probably makes sense. Please let me know if I should go ahead with this.
